lib/ClangImporter/ClangImporter.cpp

```diff
@@ -3860,21 +3964,31 @@ void ClangImporter::Implementation::lookupValue(
     // If it's a Clang declaration, try to import it.
     if (auto clangDecl = entry.dyn_cast<clang::NamedDecl *>()) {
       bool isNamespace = isa<clang::NamespaceDecl>(clangDecl);
-      Decl *realDecl =
-          importDeclReal(clangDecl->getMostRecentDecl(), CurrentVersion,
-                         /*useCanonicalDecl*/ !isNamespace);
+      Decl *realDecl = withDiagnosticTargetIfNotAlreadyReported(
+          clangDecl->getMostRecentDecl(),
+          std::function<Decl *()>([&]() -> Decl * {
```

Comment:
There was an error:
```
/Users/nuriamari/git/swift-project/swift/lib/ClangImporter/ClangImporter.cpp:2272:34: note: candidate template ignored: could not match 'function<type-parameter-0-0 ()>' against '(lambda at /Users/nuriamari/git/swift-project/swift/lib/ClangImporter/ClangImporter.cpp:3968:11)'
T ClangImporter::Implementation::withDiagnosticTargetIfNotAlreadyReported(
                                 ^
```

I will try the second approach.

Reply:
The second approach works, but it can't infer the first template parameter, I have to pass provide it explicitly. I almost prefer std::function, despite the few added characters.